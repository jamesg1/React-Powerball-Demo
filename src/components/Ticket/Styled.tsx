import styled from 'styled-components';

const Container = styled.div`
  display: flex;
  flex: 1;
  flex-direction: column;
  text-align: center;
  width: 100%;
`;

const Label = styled.div`
  width: 98%;
  font-weight: ${props => props.theme.fontWeights.bold};
  font-size: ${props => props.theme.fontSizes.xsmall};
  padding: ${props => props.theme.spacing.xsmall} 0;
  background-color: ${props => props.theme.colors.silver};
  color: ${props => props.theme.colors.white};
  text-transform: uppercase;
  margin-top: 1px;
`;

const Numbers = styled.div`
  display: flex;
  flex: 1;
  flex-wrap: wrap;
  width: 100%;
`;

export { Container, Label, Numbers };
