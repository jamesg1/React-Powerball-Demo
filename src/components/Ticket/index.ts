export { default as TicketContainer } from './TicketContainer';
