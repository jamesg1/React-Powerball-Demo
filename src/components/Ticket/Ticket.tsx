import React from 'react';
import { range } from '../../utils';
import NumberBox from '../NumberBox';
import { Container, Label, Numbers } from './Styled';

interface Props {
  label?: string;
  end: number;
  numbers: number[];
}

const Ticket: React.FC<Props> = ({ label, end, numbers }: Props) => {
  const numberRange = [...range(1, end)];
  return (
    <Container>
      {label && <Label>{label}</Label>}
      <Numbers>
        {numberRange.map(number => {
          const selected = numbers.includes(number);
          return <NumberBox key={`box-${number}`} selected={selected} number={number} />;
        })}
      </Numbers>
    </Container>
  );
};

export default Ticket;
