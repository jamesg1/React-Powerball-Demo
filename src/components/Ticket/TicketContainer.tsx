import React from 'react';
import styled from 'styled-components';
import { range } from 'utils';
import { Icon } from 'components';

const fontFamily = 'Muli, sans-serif';

const Container = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
  width: 100%;
  text-align: center;
`;

const Label = styled.div`
  width: 98%;
  font-family: ${fontFamily};
  font-weight: bold;
  font-size: 10px;
  background-color: #879bab;
  color: white;
  text-transform: uppercase;
  margin-top: 1px;
`;

const Numbers = styled.div`
  display: flex;
  flex: 1;
  width: 100%;
  flex-wrap: wrap;
`;

const purple = '#6C4796';

interface TextNumberProps {
  selected: boolean;
}

const TextNumber = styled.div<TextNumberProps>`
  position: relative;
  color: ${purple};
  font-family: ${fontFamily};
  font-weight: bold;
  background: ${props => (props.selected ? '#e9e4f1' : 'white')};
  padding: 10px;
  width: 10%;
  height: 10%;
  display: flex;
  align-items: center;
  justify-content: center;
  transition: all 400ms ease-in;
  border: 1px solid #cdcdcd;
  margin: 0 -1px -1px 0;

  > span {
    z-index: 5;
  }

  > svg {
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    margin: auto;
    width: 75%;
    height: 750%;
    position: absolute;
    fill: white;
  }
`;

interface Props {
  label?: string;
  end: number;
  numbers: number[];
}

const TicketContainer: React.FC<Props> = ({ label, end, numbers }: Props) => {
  const steps = [...range(1, end)];
  return (
    <Container>
      <Label>{label}</Label>
      <Numbers>
        {steps.map(item => {
          const selected = numbers.includes(item);
          return (
            <TextNumber selected={selected} key={item}>
              {selected && <Icon name="cross" fill="#d6cde4" />}
              <span>{item}</span>
            </TextNumber>
          );
        })}
      </Numbers>
    </Container>
  );
};

export default TicketContainer;
