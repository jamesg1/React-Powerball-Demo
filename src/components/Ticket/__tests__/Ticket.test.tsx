import React from 'react';
import Ticket from '../Ticket';
import renderWithTheme from 'styling/withTheme';
import 'jest-styled-components';

describe('<Ticket />', () => {
  test('it renders primary numbers', () => {
    const primaryNumbers = [2, 3, 10, 15, 20, 31, 35];
    const tree = renderWithTheme(<Ticket numbers={primaryNumbers} end={35} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('it renders powerball numbers', () => {
    const secondaryNumbers = [2];
    const tree = renderWithTheme(
      <Ticket numbers={secondaryNumbers} end={20} label="Select your powerball" />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
