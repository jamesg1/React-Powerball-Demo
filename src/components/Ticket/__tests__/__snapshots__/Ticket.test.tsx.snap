// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ticket /> it renders powerball numbers 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
}

.c3 {
  position: relative;
  color: #6C4796;
  font-weight: 700;
  background: #ffffff;
  padding: 10px;
  width: 10%;
  height: 10%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 400ms ease-in border:1px solid #cdcdcd;
  transition: all 400ms ease-in border:1px solid #cdcdcd;
  margin: 0 -1px -1px 0;
}

.c3 > span {
  z-index: 5;
}

.c3 > svg {
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: auto;
  width: 75%;
  height: 75%;
  position: absolute;
  fill: white;
}

.c4 {
  position: relative;
  color: #6C4796;
  font-weight: 700;
  background: #e9e4f1;
  padding: 10px;
  width: 10%;
  height: 10%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 400ms ease-in border:1px solid #cdcdcd;
  transition: all 400ms ease-in border:1px solid #cdcdcd;
  margin: 0 -1px -1px 0;
}

.c4 > span {
  z-index: 5;
}

.c4 > svg {
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: auto;
  width: 75%;
  height: 75%;
  position: absolute;
  fill: white;
}

.c1 {
  width: 98%;
  font-weight: 700;
  font-size: 10px;
  padding: 5px 0;
  background-color: #879bab;
  color: #ffffff;
  text-transform: uppercase;
  margin-top: 1px;
}

<div
  className="c0"
>
  <div
    className="c1"
    data-cy="powerball-label"
  >
    Select your powerball
  </div>
  <div
    className="c2"
  >
    <div
      className="c3"
      data-cy="box__powerball"
      selected={false}
    >
      <span>
        1
      </span>
    </div>
    <div
      className="c4"
      data-cy="box__selected__powerball"
      selected={true}
    >
      <svg
        height="100%"
        id="cross-icon"
        viewBox="0 0 24 24"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <path
          d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
          fill="#d6cde4"
        />
      </svg>
      <span>
        2
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__powerball"
      selected={false}
    >
      <span>
        3
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__powerball"
      selected={false}
    >
      <span>
        4
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__powerball"
      selected={false}
    >
      <span>
        5
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__powerball"
      selected={false}
    >
      <span>
        6
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__powerball"
      selected={false}
    >
      <span>
        7
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__powerball"
      selected={false}
    >
      <span>
        8
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__powerball"
      selected={false}
    >
      <span>
        9
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__powerball"
      selected={false}
    >
      <span>
        10
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__powerball"
      selected={false}
    >
      <span>
        11
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__powerball"
      selected={false}
    >
      <span>
        12
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__powerball"
      selected={false}
    >
      <span>
        13
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__powerball"
      selected={false}
    >
      <span>
        14
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__powerball"
      selected={false}
    >
      <span>
        15
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__powerball"
      selected={false}
    >
      <span>
        16
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__powerball"
      selected={false}
    >
      <span>
        17
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__powerball"
      selected={false}
    >
      <span>
        18
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__powerball"
      selected={false}
    >
      <span>
        19
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__powerball"
      selected={false}
    >
      <span>
        20
      </span>
    </div>
  </div>
</div>
`;

exports[`<Ticket /> it renders primary numbers 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
}

.c2 {
  position: relative;
  color: #6C4796;
  font-weight: 700;
  background: #ffffff;
  padding: 10px;
  width: 10%;
  height: 10%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 400ms ease-in border:1px solid #cdcdcd;
  transition: all 400ms ease-in border:1px solid #cdcdcd;
  margin: 0 -1px -1px 0;
}

.c2 > span {
  z-index: 5;
}

.c2 > svg {
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: auto;
  width: 75%;
  height: 75%;
  position: absolute;
  fill: white;
}

.c3 {
  position: relative;
  color: #6C4796;
  font-weight: 700;
  background: #e9e4f1;
  padding: 10px;
  width: 10%;
  height: 10%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 400ms ease-in border:1px solid #cdcdcd;
  transition: all 400ms ease-in border:1px solid #cdcdcd;
  margin: 0 -1px -1px 0;
}

.c3 > span {
  z-index: 5;
}

.c3 > svg {
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: auto;
  width: 75%;
  height: 75%;
  position: absolute;
  fill: white;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        1
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__selected"
      selected={true}
    >
      <svg
        height="100%"
        id="cross-icon"
        viewBox="0 0 24 24"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <path
          d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
          fill="#d6cde4"
        />
      </svg>
      <span>
        2
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__selected"
      selected={true}
    >
      <svg
        height="100%"
        id="cross-icon"
        viewBox="0 0 24 24"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <path
          d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
          fill="#d6cde4"
        />
      </svg>
      <span>
        3
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        4
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        5
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        6
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        7
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        8
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        9
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__selected"
      selected={true}
    >
      <svg
        height="100%"
        id="cross-icon"
        viewBox="0 0 24 24"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <path
          d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
          fill="#d6cde4"
        />
      </svg>
      <span>
        10
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        11
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        12
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        13
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        14
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__selected"
      selected={true}
    >
      <svg
        height="100%"
        id="cross-icon"
        viewBox="0 0 24 24"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <path
          d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
          fill="#d6cde4"
        />
      </svg>
      <span>
        15
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        16
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        17
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        18
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        19
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__selected"
      selected={true}
    >
      <svg
        height="100%"
        id="cross-icon"
        viewBox="0 0 24 24"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <path
          d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
          fill="#d6cde4"
        />
      </svg>
      <span>
        20
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        21
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        22
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        23
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        24
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        25
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        26
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        27
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        28
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        29
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        30
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__selected"
      selected={true}
    >
      <svg
        height="100%"
        id="cross-icon"
        viewBox="0 0 24 24"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <path
          d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
          fill="#d6cde4"
        />
      </svg>
      <span>
        31
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        32
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        33
      </span>
    </div>
    <div
      className="c2"
      data-cy="box"
      selected={false}
    >
      <span>
        34
      </span>
    </div>
    <div
      className="c3"
      data-cy="box__selected"
      selected={true}
    >
      <svg
        height="100%"
        id="cross-icon"
        viewBox="0 0 24 24"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <path
          d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
          fill="#d6cde4"
        />
      </svg>
      <span>
        35
      </span>
    </div>
  </div>
</div>
`;
